java-util/src/main/java/io/druid/java/util/common/io/smoosh/SmooshedFileMapper.java

```diff
@@ -154,6 +157,7 @@ public void close()
         }
       }
     }
+    buffersList.clear();
     Throwables.propagateIfPossible(thrown);
```

Comment:
This doesn't seem right. `Throwables.propagateIfPossible` will suppress any exception that is not a RuntimeException or Error. I think there's no reason here to only throw specific error types and suppress others, so I assume it's a bug.

Instead, it should probably be,

```java
Throwables.propagateIfPossible(thrown);
throw new RuntimeException(thrown);
```

I know this is unrelated to your patch, but it's something that looks wrong in nearby code and we could fix that too.